src/Build/BackEnd/Components/RequestBuilder/IntrinsicTasks/ItemGroupIntrinsicTask.cs

```diff
@@ -184,7 +184,20 @@ private void ExecuteAdd(ProjectItemGroupTaskItemInstance child, ItemBucket bucke
 
                 if (condition)
                 {
-                    string evaluatedValue = bucket.Expander.ExpandIntoStringLeaveEscaped(metadataInstance.Value, ExpanderOptions.ExpandAll, metadataInstance.Location, loggingContext);
+                    ExpanderOptions expanderOptions = ExpanderOptions.ExpandAll;
+                    ElementLocation location = metadataInstance.Location;
+                    if (ChangeWaves.AreFeaturesEnabled(ChangeWaves.Wave17_6) &&
```

Comment:
Since this never logs a warning anymore, we might want to rename the ExpanderOptions, and since adding a message isn't a breaking change, we probably don't need this anymore.

(And if you think we still need the changewave, please it to the doc!)

Reply:
Thanks for reminding to document!

I'd still prefer to be defensive - as the changewave guards not only the logging, but as well the heavy-lifting to detect the situation